### PR TITLE
Update SourceHandler (and related handlers) and NewsfeedHandler to use new permissions logic

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: skyportal/handlers/api/news_feed.py [hunk 1]
```diff
@@ -54,17 +53,7 @@ def get(self):
...
-                    DBSession()
-                    .query(Source.obj_id)
-                    .filter(
-                        Source.group_id.in_(
-                            [g.id for g in self.current_user.accessible_groups]
-                        )
-                    )
-                )
-            )
+            query = model.query_records_accessible_by(self.current_user)
```
Comment (acrellin) [inline skyportal/handlers/api/news_feed.py:56]: The simplicity here is a big win ✨

**Comment by dannygoldstein** [comment]
@acrellin Looks like some test failures,  perhaps the schema of the output has changed somewhat? Maybe check the `get` schema of SourceHandler before this pr and after, and verify that the same info is included in the responses?

**Comment by acrellin** [comment]
I fixed the failing tests (needed to update the data sharing API tests that make calls to the source handler), so this is green now and ready for review @dannygoldstein @kmshin1397

File: skyportal/handlers/api/source.py [hunk 2]
```diff
@@ -588,33 +604,56 @@ class Validator(Schema):
...
-                            for k, v in c.to_dict().items()
-                            if k != "attachment_bytes"
+                            **{
+                                k: v
+                                for k, v in c.to_dict().items()
+                                if k != "attachment_bytes"
+                            },
+                            "author": {
+                                **c.author.to_dict(),
+                                "gravatar_url": c.author.gravatar_url,
```
Comment (acrellin) [inline skyportal/handlers/api/source.py:637]: This is a property that isn't included in the output of `comment.to_dict()`

**Comment by acrellin** [comment]
sole test failure unrelated: `FAILED skyportal/tests/api/test_photometry.py::test_token_user_big_post`

**Review by dannygoldstein — APPROVED** [review]

**Review by kmshin1397 — APPROVED** [review]

**Pull request merged**